src/uuid-map.ts

```diff
@@ -0,0 +1,35 @@
+import * as path from 'path';
+import { Plugin } from 'obsidian';
+
+const UUID_MAP_FILENAME = 'uuids.json';
+
+export interface UuidMapStore {
+	read(): Promise<Record<string, string>>;
+	write(map: Record<string, string>): Promise<void>;
+}
+
+export class UuidMapStoreImpl implements UuidMapStore {
+	constructor(private plugin: Plugin) {}
+
+	private getUuidMapPath(): string {
+		// Default path: .obsidian/plugins/<plugin-id>/uuids.json
+		return path.join(
```

Comment:
Could we put this file in a user visible location or alternatively have a way to clear the file via the settings window? This may not come up much in a real world use case but when I was testing it was filling up with older out of date mappings and I had to go in and wipe them manually